src/main/java/neura/ui/Neura.java

```diff
@@ -0,0 +1,277 @@
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(removedTask);
+            System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            System.out.println(SEPARATOR);
+            saveTasks();  // Save tasks to file after deletion
+        } else {
+            throw new NeuraException("Invalid task number.");
+        }
+    }
+
+    // Load tasks from file
+    private void loadTasks() {
+        File dataFile = new File(FILE_PATH);
+        if (dataFile.exists()) {
+            try (BufferedReader reader = new BufferedReader(new FileReader(dataFile))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    Task task = parseTask(line);
+                    if (task != null) {
+                        tasks.add(task);
+                    }
+                }
+            } catch (IOException e) {
+                System.out.println("Error reading from file: " + e.getMessage());
+            }
+        }
+    }
+
+    // Parse a line into a Task object
+    private Task parseTask(String line) {
+        String[] parts = line.split(" \\| ");
+        try {
+            switch (parts[0]) {
+            case "T":
+                return new Todo(parts[2]);
+            case "D":
+                return new Deadline(parts[2], parts[3]);
+            case "E":
+                return new Event(parts[2], parts[3], parts[4]);
+            default:
+                return null;
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Error parsing line: " + line);
+            return null;
+        }
+    }
+
+    // Save tasks to file
+    private void saveTasks() {
+        File dataFile = new File(FILE_PATH);
+        dataFile.getParentFile().mkdirs(); // Create directory if it doesn't exist
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(dataFile))) {
+            for (Task task : tasks) {
+                writer.write(taskToFileFormat(task));
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            System.out.println("Error writing to file: " + e.getMessage());
+        }
+    }
+
+    // Convert task to file format string
+    private String taskToFileFormat(Task task) {
+        if (task instanceof Todo) {
+            return "T | " + (task.isDone() ? "1" : "0") + " | " + task.getDescription();
+        } else if (task instanceof Deadline) {
+            Deadline deadline = (Deadline) task;
+            return "D | " + (task.isDone() ? "1" : "0") + " | " + task.getDescription() + " | " + deadline.getBy();
+        } else if (task instanceof Event) {
+            Event event = (Event) task;
+            return "E | " + (task.isDone() ? "1" : "0") + " | " + task.getDescription() + " | " + event.getFrom() + " | " + event.getTo();
+        }
+        return "";
+    }
+}
```

Comment:
the main class is doing all the work. look at the increments A-Packages and A-MoreOOP and see if you can extract related methods & data to separate classes.